WooCommerce/Classes/Authentication/Epilogue/StorePickerViewController.swift

```diff
@@ -355,6 +355,7 @@ private extension StorePickerViewController {
     }
 
     func presentSiteDiscovery() {
+        ServiceLocator.authenticationManager.initialize()
```

Comment:
Excellent investigation, Huong! This works as described. 

I have a question about our fix. 

The `.initialize()` method calls `initializeWithCustomConfigs` before setting the `WordPressAuthenticator` delegate. I wonder whether it will have any side effects. i.e. Will any previously made configuration be reset? For example, I see a similar code which configures authenticator [here](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/AppCoordinator.swift#L67-L68), and it performs a few other setups. 

Would it be possible to find a central place to add the `ServiceLocator.authenticationManager.initialize()` call which will always make sure that the authenticator is iniitialized and delegate is set? (For all login flows)

What do you think?

Reply:
Hey @selanthiraiyan, thanks for sharing your concern. The code you link to is not the only place the authentication manager is configured in the `AppCoordinator`. Upon logout, the authentication manager is configured again [here](https://github.com/woocommerce/woocommerce-ios/blob/57d3faa1bc057963a8a75ffd8b280670d077e89d/WooCommerce/Classes/ViewRelated/AppCoordinator.swift#L177), so that ensures that the authenticated is set up properly for the logged-out state.

With that, calling `initialize()` should be efficient to start the site discover flow in the logged-in state IMO. Do you think that makes sense?